### PR TITLE
Remove support for JAVA_HOME

### LINKED ISSUE #55820: Should we stop honoring JAVA_HOME in favor of a dedicated environment variable
Awhile ago, we stopped honoring generic environment variables. That is environment variables that have some convention in the community (but are not actually built in to the JVM). For example, we honor our own `ES_JAVA_OPTS` over `JAVA_OPTS`. A primary reason that we do this is to avoid conflicts where a user has set some values in `JAVA_OPTS` that they wouldn't want applying to Elasticsearch. Having a dedicated environment variable avoids surprises and actually simplifies administration (since it's rare to expect there to be a common JVM flag to apply to *every* JVM that might execute on a system).

Yet, we still honor `JAVA_HOME` which has the same pitfalls, where the system could have `JAVA_HOME` configured despite a user not necessarily wanting to use the system Java with Elasticsearch.

Should we drop honoring `JAVA_HOME` to give the user explicit control over which Java is used via a dedicated environment variable `ES_JAVA_HOME` (or, the bundled JDK if it is not set)?

### PR DESCRIPTION
**Pull request by jasontedor**
This commit removes support for JAVA_HOME. As we previously deprecated usage of JAVA_HOME to override the path for the JDK, this commit follows up by removing support for JAVA_HOME. Note that we do not treat JAVA_HOME being set as a failure, as it is perfectly reasonable for a user to have JAVA_HOME configured at the system level.

Closes #55820

### CODE REVIEW

File: docs/reference/migration/migrate_8_0/packaging.asciidoc [hunk 1]
```diff
@@ -15,3 +15,17 @@ Use Java 11 or higher. Attempts to run {es} 8.0 using earlier Java versions will
...
+.JAVA_HOME is no longer supported.
+[%collapsible]
+====
+*Details* +
+`JAVA_HOME` is no longer supported to set the path for the JDK. Instead, use
+the bundled JDK (preferable), or set `ES_JAVA_HOME`.
+
+*Impact* +
+Use the bundled JDK (preferable), or set `ES_JAVA_HOME`. `JAVA_HOME` will
+silently be ignored.
```
Comment (pugnascotia) [inline docs/reference/migration/migrate_8_0/packaging.asciidoc:29]: This isn't true, the startup scripts will print a warning.

Comment (jasontedor) [reply docs/reference/migration/migrate_8_0/packaging.asciidoc:29]: Ah, would you be okay if we drop the word "silently" here?

Comment (pugnascotia) [reply docs/reference/migration/migrate_8_0/packaging.asciidoc:29]: Works for me 👍

Comment (jasontedor) [reply docs/reference/migration/migrate_8_0/packaging.asciidoc]: I pushed 7ac980778b8c5e9041b34afa95c6b5d91ae697bb.

**Review by mark-vieira — APPROVED** [review]
LGTM. Thanks, Jason!

**Pull request merged**